gunicorn/arbiter.py

```diff
@@ -203,25 +202,17 @@ def run(self):
             while True:
                 self.maybe_promote_master()
 
-                sig = self.SIG_QUEUE.pop(0) if self.SIG_QUEUE else None
-                if sig is None:
-                    self.sleep()
-                    self.murder_workers()
-                    self.manage_workers()
-                    continue
-
-                if sig not in self.SIG_NAMES:
-                    self.log.info("Ignoring unknown signal: %s", sig)
-                    continue
+                try:
+                    sig = self.SIG_QUEUE.get(timeout=1)
+                    if sig != self.WAKEUP_REQUEST:
+                        if sig != signal.SIGCHLD:
+                            self.log.info("Handling signal: %s", signal.Signals(sig).name)
+                        self.SIG_HANDLERS[sig]()
+                except queue.Empty:
+                    pass
```

Comment:
```suggestion
                timeout = 1
                while True:
                    try:
                        sig = self.SIG_QUEUE.get(timeout=timeout)
                    except queue.Empty:
                        break
                    timeout = 0
                    if sig != self.WAKEUP_REQUEST:
                        self.log.info("Handling signal: %s", signal.Signals(sig).name)
                        self.SIG_HANDLERS[sig]()
```

Can you consider some of the suggested changes?:

1. Log the info message unconditionally for all signals.
1. Limit the try block.
1. Make sure we go through all queued signals (without waiting for subsequent ones) before .murder_workers() is called.

Reply:
Sure! I did as follows:

1. Made so that SIGCHLD gets logged with debug severity. In the normal case, I think logging with info severity gets too spammy.
2. Sure! I refactored getting the list of signals into its own method, this way I found it easier to limit the scope.
3. Same here, solved by getting a list of signals from a method instead.